lib/xilinx/xc7series/bitstream_writer.cc

```diff
@@ -0,0 +1,224 @@
+/*
+ * TODO
+ * -Finish type 1/2 support
+ * -Review sample bitstream padding. What are they for?
+ */
+#include <prjxray/xilinx/xc7series/bitstream_writer.h>
+
+#include <prjxray/bit_ops.h>
+
+namespace prjxray {
+namespace xilinx {
+namespace xc7series {
+
+// Per UG470 pg 80: Bus Width Auto Detection
+std::array<uint32_t, 6> BitstreamWriter::header{
+    0xFFFFFFFF, 0x000000BB, 0x11220044, 0xFFFFFFFF, 0xFFFFFFFF, 0xAA995566};
+
+/**************************************************
+ * BitstreamWriter::BitstreamWriter
+ *************************************************/
+
+BitstreamWriter::BitstreamWriter(const packets_t& packets) : packets(packets) {}
+
+/**************************************************
+ *
+ *************************************************/
+
+BitstreamWriter::packet_iterator BitstreamWriter::iterator::packet_begin() {
+	// itr_packets = packets.begin();
+	const ConfigurationPacket& packet = *itr_packets;
+
+	return BitstreamWriter::packet_iterator(
+	    &packet, BitstreamWriter::packet_iterator::STATE_BEGIN,
```

Comment:
How does STATE_BEGIN get changed to STATE_HEADER?  I'm not seeing how this works.

Reply:
STATE_BEGIN = 1,
			STATE_HEADER = 1,
I can eliminate STATE_BEGIN if that is confusing